src/ragapp/backend/routers/chat/index.py

```diff
@@ -38,15 +43,39 @@ async def chat(
         logger.info(
             f"Creating chat engine with filters: {str(filters)}",
         )
-        chat_engine = get_chat_engine(filters=filters, params=params)
-
         event_handler = EventCallbackHandler()
-        chat_engine.callback_manager.handlers.append(event_handler)  # type: ignore
-
-        response = await chat_engine.astream_chat(last_message_content, messages)
-        process_response_nodes(response.source_nodes, background_tasks)
+        chat_engine = create_chat_engine(
+            filters=filters,
+            params=params,
+            event_handlers=[event_handler],
+            chat_history=messages,
+        )
 
-        return VercelStreamResponse(request, event_handler, response, data)
+        if isinstance(chat_engine, CondensePlusContextChatEngine) or isinstance(
+            chat_engine, AgentRunner
+        ):
+            event_handler = EventCallbackHandler()
+            chat_engine.callback_manager.handlers.append(event_handler)  # type: ignore
+
+            response = await chat_engine.astream_chat(last_message_content, messages)
+            process_response_nodes(response.source_nodes, background_tasks)
+
+            return ContextEngineVercelStreamResponse(
+                request=request,
+                chat_data=data,
+                event_handler=event_handler,
+                response=response,
+            )
+        else:
+            task = asyncio.create_task(
+                chat_engine.run(input=last_message_content, streaming=True)
+            )
+            return AgentsVercelStreamResponse(
+                request=request,
+                chat_data=data,
+                task=task,
+                events=chat_engine.stream_events(),  # Call the method to get the generator
+            )
```

Comment:
how about this naming:
```suggestion
            return ChatEngineVercelStreamResponse(
                request=request,
                chat_data=data,
                event_handler=event_handler,
                response=response,
            )
        else:
            task = asyncio.create_task(
                chat_engine.run(input=last_message_content, streaming=True)
            )
            return WorkflowVercelStreamResponse(
                request=request,
                chat_data=data,
                task=task,
                events=chat_engine.stream_events(),  # Call the method to get the generator
            )
```